pkg/operator/controllers/imageconfig/image_controller_test.go

```diff
@@ -205,35 +196,73 @@ func TestImageConfigReconciler(t *testing.T) {
 			if err != nil && err.Error() != tt.wantErr ||
 				err == nil && tt.wantErr != "" {
 				t.Error(err)
-				t.Error("----------------------")
-				t.Error(tt.wantErr)
 			}
 			imgcfg, err := r.configcli.ConfigV1().Images().Get(ctx, request.Name, metav1.GetOptions{})
 			if err != nil {
 				t.Fatal(err)
 			}
-			imgcfgjson := getRegistrySources(imgcfg)
 
-			if string(imgcfgjson) != strings.TrimSpace(tt.wantConfig) && tt.wantConfig != "" {
-				t.Error(string(imgcfgjson))
-				t.Error("----------------------")
-				t.Error(tt.wantConfig)
+			if !reflect.DeepEqual(imgcfg.Spec.RegistrySources, tt.wantRegistrySources) {
+				t.Error(cmp.Diff(imgcfg.Spec.RegistrySources, tt.wantRegistrySources))
 			}
-
 		})
 	}
 }
 
-func getRegistrySources(imgcfg *configv1.Image) []byte {
-	var registrySourceJSON []byte
-	if imgcfg.Spec.RegistrySources.AllowedRegistries != nil {
-		imgRegistries := imgcfg.Spec.RegistrySources.AllowedRegistries
-		sort.Strings(imgRegistries)
-		registrySourceJSON, _ = json.Marshal(imgRegistries)
-	} else {
-		imgRegistries := imgcfg.Spec.RegistrySources.BlockedRegistries
-		sort.Strings(imgRegistries)
-		registrySourceJSON, _ = json.Marshal(imgRegistries)
+func TestGetCloudAwareRegistries(t *testing.T) {
+	type test struct {
+		name       string
+		instance   *arov1alpha1.Cluster
+		wantResult []string
+		wantErr    string
+	}
+
+	for _, tt := range []*test{
+		{
+			name: "public cloud",
+			instance: &arov1alpha1.Cluster{
+				Spec: arov1alpha1.ClusterSpec{
+					ACRDomain:     "arointsvc.azurecr.io",
+					AZEnvironment: azureclient.PublicCloud.Environment.Name,
+					Location:      "eastus",
+				},
+			},
+			wantResult: []string{"arointsvc.azurecr.io", "arosvc.eastus.data.azurecr.io"},
+		},
+		{
+			name: "us gov cloud",
+			instance: &arov1alpha1.Cluster{
+				Spec: arov1alpha1.ClusterSpec{
+					ACRDomain:     "arointsvc.azurecr.us",
+					AZEnvironment: azureclient.USGovernmentCloud.Environment.Name,
+					Location:      "eastus",
+				},
+			},
+			wantResult: []string{"arointsvc.azurecr.us", "arosvc.eastus.data.azurecr.us"},
+		},
+		{
+			name: "unsupported cloud",
+			instance: &arov1alpha1.Cluster{
+				Spec: arov1alpha1.ClusterSpec{
+					ACRDomain:     "arointsvc.azurecr.io",
+					AZEnvironment: "FakeCloud",
+					Location:      "eastus",
+				},
+			},
+			wantErr: "cloud environment FakeCloud is not supported",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := getCloudAwareRegistries(tt.instance)
+
+			if err != nil && err.Error() != tt.wantErr ||
```

Comment:
makes sense :+1:, but I might've stared at this longer than most to understand each condition :sweat_smile: 

if no unexpected errors occurred unless the error is the error we want, or no errors occurred and we wanted an error, have the test return error.

Reply:
This is the very common pattern we use across the codebase. But I agree - readability it not the best.